### PR TITLE
Add GetActorFromUserId to actor system

### PR DESCRIPTION
**Pull request by Chief-Engineer**
for space-wizards/space-station-14#18709

### CODE REVIEW

File: Robust.Server/GameObjects/EntitySystems/ActorSystem.cs [hunk 1]
```diff
@@ -116,6 +120,19 @@ private void OnActorShutdown(EntityUid entity, ActorComponent component, Compone
             // The player is fully detached now that the component has shut down.
             RaiseLocalEvent(entity, new PlayerDetachedEvent(entity, component.PlayerSession), true);
         }
+
+        public (IPlayerSession? actor, EntityUid? actorEntity) GetActorFromUserId(NetUserId? userId)
```
Comment (EmoGarbage404) [inline Robust.Server/GameObjects/EntitySystems/ActorSystem.cs]: feels more sane to do a TryGet that returns a boolean and has out variables for the actor and actor entity.

Comment (Chief-Engineer) [reply Robust.Server/GameObjects/EntitySystems/ActorSystem.cs]: Would it return true or false if it was able to get a player session but no controlled entity

Comment (metalgearsloth) [reply Robust.Server/GameObjects/EntitySystems/ActorSystem.cs]: True for IPlayerSession not being null, then annotate both appropriately (i.e. notnullwhentrue for IPlayerSession and maybenullwhentrue for the entity).

File: Robust.Server/GameObjects/EntitySystems/ActorSystem.cs [hunk 2]
```diff
@@ -116,6 +121,19 @@ private void OnActorShutdown(EntityUid entity, ActorComponent component, Compone
...
         }
+
+        public bool TryGetActorFromUserId(NetUserId? userId, [NotNullWhen(true)] out IPlayerSession? actor, [MaybeNullWhen(true)] out EntityUid? actorEntity)
+        {
+            actor = null;
+            actorEntity = null;
+            if (userId != null)
+            {
+                actor = _playerManager.GetSessionByUserId(userId.Value);
+                actorEntity = actor.AttachedEntity;
```
Comment (ElectroJr) [inline Robust.Server/GameObjects/EntitySystems/ActorSystem.cs]: This should use `TryGetSessionById`

**Pull request merged**